tasks/polarisTask/util/PolarisPlatformSupport.ts

```diff
@@ -6,7 +6,22 @@ export default class PolarisPlatformSupport {
         if (platform == "win32") {
             return "/api/tools/" + client + "_cli-win64.zip";
         } else if (platform == "darwin") {
-            return "/api/tools/" + client + "_cli-macosx.zip";
+            // The mac runner architecture in the ADO returns same for both arm and non arm machines.
+            // So, we fetch the CPU information and try to detect if it is a non-arm runner
+            const cpuInfo = os.cpus();
+            if ((cpuInfo.length) > 0) {
+                const isIntel = cpuInfo[0].model.includes("Intel");
+                if (isIntel) {
+                    return "/api/tools/" + client + "_cli-macosx.zip";
+                }
+                return "/api/tools/" + client + "_cli-macos_arm.zip";
+            } else {
+                const arch = os.arch()
+                if (arch == "x86_64") {
```

Comment:
minor: may we could do `if (os.arch() == "x86_64") {` instead of extra const declaration I think